--- configure.ac.orig	2016-10-16 11:27:36.000000000 -0400
+++ configure.ac	2016-10-16 11:28:48.000000000 -0400
@@ -135,8 +135,8 @@
 
 dnl This must stay after AC_CANONICAL_HOST, which defines $host.
 case "$host" in
-*openbsd*|*freebsd*)
-	dnl Do not set '-Wl,--no-undefined' on freebsd/openbsd
+*openbsd*|*freebsd*|*darwin*)
+	dnl Do not set '-Wl,--no-undefined' on freebsd/openbsd/darwin
 	;;
 *)
 	LDFLAGS="$LDFLAGS -Wl,--no-undefined"
