--- configure.ac.orig	2015-10-27 09:57:36.000000000 -0700
+++ configure.ac	2015-10-27 09:58:30.000000000 -0700
@@ -104,10 +104,10 @@
   gsettings-desktop-schemas >= $GSETTINGS_DESKTOP_SCHEMAS_REQUIRED
 ])
 
-PKG_CHECK_MODULES([BLUETOOTH_APPLET], [
-  gtk+-3.0 >= $GTK_REQUIRED
-  gnome-bluetooth-1.0
-])
+dnl PKG_CHECK_MODULES([BLUETOOTH_APPLET], [
+dnl   gtk+-3.0 >= $GTK_REQUIRED
+dnl   gnome-bluetooth-1.0
+dnl ])
 
 PKG_CHECK_MODULES([COMMON], [
   gtk+-3.0 >= $GTK_REQUIRED
--- gnome-flashback/Makefile.am.orig	2015-10-27 10:00:56.000000000 -0700
+++ gnome-flashback/Makefile.am	2015-10-27 10:01:26.000000000 -0700
@@ -2,7 +2,6 @@
 
 SUBDIRS = \
 	libautomount-manager \
-	libbluetooth-applet \
 	libcommon \
 	libdesktop-background \
 	libdisplay-config \
@@ -47,7 +46,6 @@
 gnome_flashback_LDADD = \
 	$(GNOME_FLASHBACK_LIBS) \
 	$(top_builddir)/gnome-flashback/libautomount-manager/libautomount-manager.la \
-	$(top_builddir)/gnome-flashback/libbluetooth-applet/libbluetooth-applet.la \
 	$(top_builddir)/gnome-flashback/libdesktop-background/libdesktop-background.la \
 	$(top_builddir)/gnome-flashback/libdisplay-config/libdisplay-config.la \
 	$(top_builddir)/gnome-flashback/libend-session-dialog/libend-session-dialog.la \
