--- data/Makefile.in.orig	2009-07-28 08:56:29.000000000 -0700
+++ data/Makefile.in	2009-07-30 16:52:08.000000000 -0700
@@ -470,7 +470,7 @@
 @INTLTOOL_DESKTOP_RULE@
 
 install-icons:			
-	for size in 16x16 22x22 24x24 32x32 48x48 scalable; do mkdir --parents $(DESTDIR)/$(datadir)/icons/hicolor/$$size/apps; done
+	for size in 16x16 22x22 24x24 32x32 48x48 scalable; do mkdir -p $(DESTDIR)/$(datadir)/icons/hicolor/$$size/apps; done
 
 	install -m 644 $(srcdir)/pessulus-16.png $(DESTDIR)/$(datadir)/icons/hicolor/16x16/apps/pessulus.png 
 	install -m 644 $(srcdir)/pessulus-22.png $(DESTDIR)/$(datadir)/icons/hicolor/22x22/apps/pessulus.png 
