--- Makefile.in.orig	2009-07-28 08:56:30.000000000 -0700
+++ Makefile.in	2009-07-30 16:50:23.000000000 -0700
@@ -275,7 +275,7 @@
 top_srcdir = @top_srcdir@
 NULL = 
 SUBDIRS = data po Pessulus
-bin_SCRIPTS = pessulus
+bin_SCRIPTS = pessulus.py
 EXTRA_DIST = \
 	HACKING			\
 	MAINTAINERS		\
@@ -286,7 +286,7 @@
 	$(NULL)
 
 CLEANFILES = \
-	pessulus
+	pessulus.py
 
 DISTCLEANFILES = \
         intltool-extract	\
@@ -812,7 +812,7 @@
 	uninstall-binSCRIPTS
 
 
-pessulus: pessulus.in Makefile
+pessulus.py: pessulus.in Makefile
 	$(AM_V_GEN)sed -e s!\@pythondir\@!@pythondir@! -e s!\@PYTHON\@!@PYTHON@! < $< > $@
 	$(AM_V_at)chmod a+x $@
 
