--- gkr/gkr-backend.vala.orig	2013-11-25 15:34:04.000000000 -0800
+++ gkr/gkr-backend.vala	2013-11-25 15:38:44.000000000 -0800
@@ -64,7 +64,7 @@
 	private Gtk.ActionGroup _actions;
 
 	construct {
-		return_if_fail(_instance == null);
+		return_val_if_fail(_instance == null, null);
 		Backend._instance = this;
 
 		this._actions = BackendActions.instance(this);
@@ -154,7 +154,7 @@
 	}
 
 	public static Backend instance() {
-		return_if_fail(Backend._instance != null);
+		return_val_if_fail(Backend._instance != null, null);
 		return Backend._instance;
 	}
 
