--- glom/libglom/connectionpool.cc.orig	2015-02-10 03:00:11.000000000 -0800
+++ glom/libglom/connectionpool.cc	2015-07-18 03:21:41.000000000 -0700
@@ -605,7 +605,7 @@
 // TODO: This is probably mingw specific
 static __p_sig_fn_t previous_sig_handler = SIG_DFL;
 #else
-static sighandler_t previous_sig_handler = SIG_DFL; /* Arbitrary default */
+static sig_t previous_sig_handler = SIG_DFL; /* Arbitrary default */
 #endif
 
 /* This is a Linux/Unix signal handler,
