--- src/goabackend/goaproviderfactory.c.orig	2014-01-08 07:53:27.000000000 -0800
+++ src/goabackend/goaproviderfactory.c	2014-01-21 15:16:14.000000000 -0800
@@ -86,7 +86,7 @@
 {
   GoaProviderFactoryClass *klass;
 
-  g_return_if_fail (GOA_IS_PROVIDER_FACTORY (factory));
+  g_return_val_if_fail (GOA_IS_PROVIDER_FACTORY (factory), NULL);
 
   klass = GOA_PROVIDER_FACTORY_GET_CLASS (factory);
   g_return_if_fail (klass->get_providers != NULL);
