--- src/tracker-extract/tracker-extract-libav.c.orig	2014-03-24 21:58:01.000000000 +0100
+++ src/tracker-extract/tracker-extract-libav.c	2014-03-24 22:12:31.000000000 +0100
@@ -19,6 +19,8 @@
  */
 
 
+#include "config.h"
+
 #include <glib.h>
 
 #include <libtracker-common/tracker-ontologies.h>
@@ -26,7 +28,9 @@
 
 #include <libtracker-extract/tracker-extract.h>
 
-#include <tracker-media-art.h>
+#ifdef HAVE_LIBMEDIAART
+#include <libmediaart/mediaart.h>
+#endif
 
 #include <libavcodec/avcodec.h>
 #include <libavformat/avformat.h>
