--- src/libtracker-sparql/Makefile.am.orig	2015-07-05 03:32:27.000000000 -0700
+++ src/libtracker-sparql/Makefile.am	2015-07-18 03:53:06.000000000 -0700
@@ -90,7 +90,7 @@
 
 if HAVE_INTROSPECTION
 Tracker-@TRACKER_API_VERSION@.typelib: Tracker-$(TRACKER_API_VERSION).gir
-	@INTROSPECTION_COMPILER@ --shared-library=libtracker-sparql-@TRACKER_API_VERSION@.so.0 -o $@ $^
+	@INTROSPECTION_COMPILER@ --shared-library=${prefix}/lib/libtracker-sparql-@TRACKER_API_VERSION@.dylib -o $@ $^
 
 girdir = $(datadir)/gir-1.0
 gir_DATA = Tracker-$(TRACKER_API_VERSION).gir
