--- src/libtracker-miner/tracker-file-notifier.c.orig	2015-07-21 17:29:39.000000000 -0700
+++ src/libtracker-miner/tracker-file-notifier.c	2015-07-21 17:32:40.000000000 -0700
@@ -1767,8 +1767,8 @@
 	TrackerFileNotifierPrivate *priv;
 	GFile *canonical;
 
-	g_return_if_fail (TRACKER_IS_FILE_NOTIFIER (notifier));
-	g_return_if_fail (G_IS_FILE (file));
+	g_return_val_if_fail (TRACKER_IS_FILE_NOTIFIER (notifier), G_FILE_TYPE_UNKNOWN);
+	g_return_val_if_fail (G_IS_FILE (file), G_FILE_TYPE_UNKNOWN);
 
 	priv = notifier->priv;
 	canonical = tracker_file_system_get_file (priv->file_system,
