From 5e6b1912b0ed627dccb16b2cf00547cfeeb7654f Mon Sep 17 00:00:00 2001
From: Murray Cumming <murrayc@murrayc.com>
Date: Thu, 17 Dec 2015 20:01:12 +0100
Subject: Gtk::Border: Fix the build now that GtkBorder is a just a typedef.

Bug #759603 (Matthias Clasen)
---
 gtk/src/border.hg | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/gtk/src/border.hg b/gtk/src/border.hg
index 1940cb6..ad5064a 100644
--- gtk/src/border.hg
+++ gtk/src/border.hg
@@ -19,6 +19,11 @@
 _DEFS(gtkmm,gtk)
 
 #include <glibmm/value.h>
+#include <gdk/gdk.h>
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+extern "C" { typedef GdkBorder GtkBorder; }
+#endif /* DOXYGEN_SHOULD_SKIP_THIS */
 
 namespace Gtk
 {
@@ -28,6 +33,7 @@ namespace Gtk
 class Border
 {
   _CLASS_BOXEDTYPE(Border, GtkBorder, gtk_border_new, gtk_border_copy, gtk_border_free)
+  _CUSTOM_STRUCT_PROTOTYPE
 public:
 
   /** Get the width of the left border.
-- 
cgit v0.11.2

