--- gedit/gedit-close-confirmation-dialog.c.orig	2014-06-20 02:28:00.000000000 -0700
+++ gedit/gedit-close-confirmation-dialog.c	2014-06-20 02:28:11.000000000 -0700
@@ -153,8 +153,7 @@
 
 	action_area = gtk_dialog_get_action_area (GTK_DIALOG (dlg));
 	gtk_button_box_set_layout (GTK_BUTTON_BOX (action_area), GTK_BUTTONBOX_EXPAND);
-	gtk_style_context_add_class (gtk_widget_get_style_context (GTK_WIDGET (dlg)),
-	                                                           GTK_STYLE_CLASS_MESSAGE_DIALOG);
+	gtk_style_context_add_class (gtk_widget_get_style_context (GTK_WIDGET (dlg)), "prompt");
 
 	g_signal_connect (dlg,
 			  "response",
