--- glom/main.cc.orig	2013-04-19 13:05:43.000000000 -0700
+++ glom/main.cc	2013-04-19 13:07:41.000000000 -0700
@@ -372,7 +372,7 @@
 } //namespace Glom
 
 #ifndef G_OS_WIN32
-extern "C" void __libc_freeres(void);
+//extern "C" void __libc_freeres(void);
 #endif
 
 int
@@ -381,7 +381,7 @@
 #ifndef G_OS_WIN32
   //Force some cleanup at exit,
   //to help valgrind to detect memory leaks:
-  atexit(__libc_freeres);
+  //atexit(__libc_freeres);
 #else
   WSADATA data;
   int errcode = WSAStartup(MAKEWORD(2, 0), &data);
