--- glom/libglom/connectionpool.cc.orig	2011-04-15 18:21:22.000000000 -0700
+++ glom/libglom/connectionpool.cc	2011-04-15 18:23:33.000000000 -0700
@@ -600,7 +600,7 @@
 // TODO: This is probably mingw specific
 static __p_sig_fn_t previous_sig_handler = SIG_DFL;
 #else
-static sighandler_t previous_sig_handler = SIG_DFL; /* Arbitrary default */
+static sig_t previous_sig_handler = SIG_DFL; /* Arbitrary default */
 #endif
 
 /* This is a Linux/Unix signal handler,
