diff -ur tests.orig/test_python_module.cc tests/test_python_module.cc
--- tests.orig/test_python_module.cc	2011-07-05 23:56:40.000000000 -0700
+++ tests/test_python_module.cc	2013-04-19 02:04:25.000000000 -0700
@@ -6,7 +6,7 @@
 #include "glom/python_embed/glom_python.h"
 
 #ifndef G_OS_WIN32
-extern "C" void __libc_freeres(void);
+//extern "C" void __libc_freeres(void);
 #endif
 
 namespace Glom
@@ -42,7 +42,7 @@
 int main ()
 {
 #ifndef G_OS_WIN32
-  atexit(__libc_freeres);
+  //atexit(__libc_freeres);
 #endif
   Glom::libglom_init();  // Calls PyInitialize()
   
