--- src/goabackend/goaproviderfactory.c.orig	2013-11-27 07:11:13.000000000 -0800
+++ src/goabackend/goaproviderfactory.c	2013-11-27 07:13:03.000000000 -0800
@@ -114,7 +114,7 @@
 {
   GoaProviderFactoryClass *klass;
 
-  g_return_if_fail (GOA_IS_PROVIDER_FACTORY (factory));
+  g_return_val_if_fail (GOA_IS_PROVIDER_FACTORY (factory), NULL);
 
   klass = GOA_PROVIDER_FACTORY_GET_CLASS (factory);
   return klass->get_providers_finish (factory, out_providers, result, error);
--- src/goabackend/goatelepathyfactory.c.orig	2013-11-27 07:19:01.000000000 -0800
+++ src/goabackend/goatelepathyfactory.c	2013-11-27 07:19:36.000000000 -0800
@@ -50,7 +50,7 @@
 get_provider (GoaProviderFactory *factory,
               const gchar        *provider_name)
 {
-  g_return_if_fail (GOA_IS_TELEPATHY_FACTORY (factory));
+  g_return_val_if_fail (GOA_IS_TELEPATHY_FACTORY (factory), NULL);
 
   return GOA_PROVIDER (goa_telepathy_provider_new_from_protocol_name (provider_name));
 }
