--- meson.build.orig	2016-11-22 06:10:06.000000000 -0800
+++ meson.build	2016-11-26 02:06:07.000000000 -0800
@@ -184,7 +184,9 @@
 endif
 
 # Required dependencies
-mathlib = cc.find_library('m', required: false)
+# mathlib = cc.find_library('m', required: false)
+# find_library claims to have found libm but it doesn't really exist
+mathlib = cc.find_library('xxx', required: false)
 threadlib = dependency('threads', required: true)
 
 # Optional dependency on GObject
