--- configure.ac.orig	2014-01-08 12:49:10.000000000 -0800
+++ configure.ac	2014-01-21 22:10:12.000000000 -0800
@@ -317,8 +317,8 @@
 
 dnl This must appear after AC_CANONICAL_HOST, which defines $host.
 case "$host" in
-*openbsd*|*freebsd*)
-	dnl Do not set '-Wl,--no-undefined' on freebsd/openbsd
+*openbsd*|*freebsd*|*darwin*)
+	dnl Do not set '-Wl,--no-undefined' on freebsd/openbsd/darwin
 	;;
 *)
 	LDFLAGS="$LDFLAGS -Wl,--no-undefined"
