--- configure.ac.orig	2015-02-10 08:19:57.000000000 -0800
+++ configure.ac	2015-02-19 17:39:19.000000000 -0800
@@ -219,7 +219,7 @@
 	-Wnested-externs"
 
 proposed_cxx_warning_flags="$proposed_warning_flags
-	-Wabi -Wnoexcept"
+	-Wabi"
 
 AS_COMPILER_FLAGS(WARNING_FLAGS, [$proposed_c_warning_flags])
 AC_SUBST(WARNING_FLAGS)
@@ -328,8 +328,8 @@
 
 dnl This must appear after AC_CANONICAL_HOST, which defines $host.
 case "$host" in
-*openbsd*|*freebsd*)
-	dnl Do not set '-Wl,--no-undefined' on freebsd/openbsd
+*openbsd*|*freebsd*|*darwin*)
+	dnl Do not set '-Wl,--no-undefined' on freebsd/openbsd/darwin
 	;;
 *)
 	LDFLAGS="$LDFLAGS -Wl,--no-undefined"
@@ -634,8 +634,8 @@
 
 if test "x$with_libdb" != "xno" ; then
 	if test "x$with_libdb" != "xyes" ; then
-		DB_CFLAGS="-I$with_libdb/include"
-		DB_LIBS="-L$with_libdb/lib -ldb"
+		DB_CFLAGS="-I$prefix/include/db46"
+		DB_LIBS="-L$prefix/lib/db46 -ldb"
 	else
 		if test -z "$DB_CFLAGS" -a -z "$DB_LIBS"; then
 			DB_CFLAGS=""
