diff -ur src.orig/sp-lpe-item.cpp src/sp-lpe-item.cpp
--- src.orig/sp-lpe-item.cpp	2014-06-24 12:03:08.000000000 -0700
+++ src/sp-lpe-item.cpp	2014-06-24 12:03:21.000000000 -0700
@@ -50,6 +50,8 @@
 #include <algorithm>
 
 /* LPEItem base class */
+static void sp_lpe_item_enable_path_effects(SPLPEItem *lpeitem, bool enable);
+
 static void lpeobject_ref_modified(SPObject *href, guint flags, SPLPEItem *lpeitem);
 
 static void sp_lpe_item_create_original_path_recursive(SPLPEItem *lpeitem);
@@ -113,7 +115,7 @@
                 this->current_path_effect = NULL;
 
                 // Disable the path effects while populating the LPE list
-                enablePathEffects(false);
+                 sp_lpe_item_enable_path_effects(this, false);
 
                 // disconnect all modified listeners:
                 for ( std::list<sigc::connection>::iterator mod_it = this->lpe_modified_connection_list->begin();
@@ -166,7 +168,7 @@
                     }
                 }
 
-                enablePathEffects(true);
+                sp_lpe_item_enable_path_effects(this, true);
             }
             break;
 
@@ -407,7 +409,7 @@
         sp_lpe_item_update_patheffect(this, false, true);
 
         // Disable the path effects while preparing the new lpe
-        enablePathEffects(false);
+        sp_lpe_item_enable_path_effects(this, false);
 
         // Add the new reference to the list of LPE references
         HRefList hreflist;
@@ -444,7 +446,7 @@
         }
 
         //Enable the path effects now that everything is ready to apply the new path effect
-        enablePathEffects(true);
+        sp_lpe_item_enable_path_effects(this, true);
 
         // Apply the path effect
         sp_lpe_item_update_patheffect(this, true, true);
@@ -940,6 +942,24 @@
     return forked;
 }
 
+// Enable or disable the path effects of the item.
+// The counter allows nested calls
+static void sp_lpe_item_enable_path_effects(SPLPEItem *lpeitem, bool enable)
+{
+    if (enable) {
+        lpeitem->path_effects_enabled++;
+    }
+    else {
+        lpeitem->path_effects_enabled--;
+    }
+}
+
+// Are the path effects enabled on this item ?
+bool SPLPEItem::pathEffectsEnabled() const
+{
+    return path_effects_enabled > 0;
+}
+
 /*
   Local Variables:
   mode:c++
diff -ur src.orig/sp-lpe-item.h src/sp-lpe-item.h
--- src.orig/sp-lpe-item.h	2014-06-24 12:03:08.000000000 -0700
+++ src/sp-lpe-item.h	2014-06-24 12:03:21.000000000 -0700
@@ -39,13 +39,11 @@
 typedef std::list<Inkscape::LivePathEffect::LPEObjectReference *> PathEffectList;
 
 class SPLPEItem : public SPItem {
-private:
-    mutable bool path_effects_enabled;  // (mutable because preserves logical const-ness)
-
 public:
-  SPLPEItem();
-  virtual ~SPLPEItem();
+	SPLPEItem();
+	virtual ~SPLPEItem();
 
+    int path_effects_enabled;
 
     PathEffectList* path_effect_list;
     std::list<sigc::connection> *lpe_modified_connection_list; // this list contains the connections for listening to lpeobject parameter changes
@@ -74,8 +72,7 @@
 
     bool performPathEffect(SPCurve *curve);
 
-    void enablePathEffects(bool enable) const { path_effects_enabled = enable; }; // (const because logically const)
-    bool pathEffectsEnabled() const { return path_effects_enabled; };
+    bool pathEffectsEnabled() const;
     bool hasPathEffect() const;
     bool hasPathEffectOfType(int const type) const;
     bool hasPathEffectRecursive() const;
