--- glom/frame_glom.cc.orig	2015-07-21 01:03:05.000000000 -0700
+++ glom/frame_glom.cc	2015-07-21 01:04:25.000000000 -0700
@@ -1628,7 +1628,7 @@
     Utils::get_glade_widget_derived_with_warning(m_pDialog_Relationships);
     if(!m_pDialog_Relationships)
     {
-      std::cerr << G_STRFUNC << ": m_pDialog_Relationships is null." << std::cerr;
+      std::cerr << G_STRFUNC << ": m_pDialog_Relationships is null." << std::endl;
       return;
     }
     
@@ -1699,7 +1699,7 @@
       Utils::get_glade_child_widget_derived_with_warning(m_pBox_Reports);
     if(!m_pBox_Reports)
     {
-      std::cerr << G_STRFUNC << ": m_pBox_Reports is null." << std::cerr;
+      std::cerr << G_STRFUNC << ": m_pBox_Reports is null." << std::endl;
       return;
     }
 
@@ -1990,7 +1990,7 @@
     Utils::get_glade_widget_derived_with_warning(m_pDialogConnection);
     if(!m_pDialogConnection)
     {
-      std::cerr << G_STRFUNC << ": m_pBox_Reports is null." << std::cerr;
+      std::cerr << G_STRFUNC << ": m_pBox_Reports is null." << std::endl;
       return false;
     }
 
@@ -2394,7 +2394,7 @@
     Utils::get_glade_widget_with_warning("glom_developer.glade", "dialog_error_create_database", dialog);
     if(!dialog)
     {
-      std::cerr << G_STRFUNC << ": dialog is null." << std::cerr;
+      std::cerr << G_STRFUNC << ": dialog is null." << std::endl;
       return false;
     }
 
