--- configure.ac.orig	2015-11-27 01:21:08.000000000 -0800
+++ configure.ac	2015-12-05 15:34:57.000000000 -0800
@@ -159,7 +159,10 @@
 
 
 # Libraries used by libglom:
-REQUIRED_LIBGLOM_LIBS='giomm-2.4 >= 2.46.1 libxml++-3.0 >= 2.24.0 libxslt >= 1.1.10 pygobject-3.0 >= 2.29.0 libgdamm-5.0 >= 4.99.10 libgda-5.0 >= 5.2.1 libgda-postgres-5.0 libgda-postgres-5.0 libgda-mysql-5.0 libarchive >= 3.0'
+REQUIRED_LIBGLOM_LIBS='giomm-2.4 >= 2.46.1 libxml++-3.0 >= 2.24.0 libxslt >= 1.1.10 pygobject-3.0 >= 2.29.0 libgdamm-5.0 >= 4.99.10 libgda-5.0 >= 5.2.1 libgda-postgres-5.0 libgda-postgres-5.0 libarchive >= 3.0'
+
+AS_IF([test "x$glom_enable_mysql" = xyes],
+      [REQUIRED_LIBGLOM_LIBS="$REQUIRED_LIBGLOM_LIBS libgda-mysql-5.0"])
 
 AS_IF([test "x$glom_host_win32" != xyes],
       [REQUIRED_LIBGLOM_LIBS="$REQUIRED_LIBGLOM_LIBS libepc-1.0 >= 0.4.0"])
@@ -232,6 +235,17 @@
 # so we can use pyexec_LTLIBRARIES in the .am files:
 AM_PATH_PYTHON
 
+dnl Override standard python locations with MacPorts version specific ones
+
+AC_SUBST([PYTHON_PREFIX], [`$PYTHON -c 'import sys; print(sys.prefix);'`])
+AC_SUBST([PYTHON_EXEC_PREFIX], [`$PYTHON -c 'import sys; print(sys.exec_prefix)'`])
+
+am_cv_python_pythondir=$PYTHON_PREFIX/lib/python$PYTHON_VERSION/site-packages
+am_cv_python_pyexecdir=$PYTHON_EXEC_PREFIX/lib/python$PYTHON_VERSION/site-packages
+
+AC_SUBST([pythondir], [$am_cv_python_pythondir])
+AC_SUBST([pyexecdir], [$am_cv_python_pyexecdir])
+
 # Get the compiler and linker flags for embedding Python.
 # To specify a particular python version you must pass PYTHON_VERSION.
 # For instance, ./configure PYTHON_VERSION="3.4" --prefix=/opt/gnome
