--- configure.ac.orig	2009-06-29 07:12:46.000000000 -0700
+++ configure.ac	2009-06-29 07:14:21.000000000 -0700
@@ -342,12 +342,10 @@
 #include <iconv.h>
 #include <stdlib.h>],[
 
-int main (int argc, char **argv)
-{
 	iconv_t cd;
 	cd = iconv_open ("UTF-8", "ISO-8859-1");
 	exit (0);
-}
+
 ], ac_cv_libiconv=yes, ac_cv_libiconv=no))
 if test $ac_cv_libiconv = yes; then
 	ICONV_LIBS="$ICONV_LIBS -liconv"
