--- Source/cmake/OptionsGTK.cmake.orig	2016-03-08 10:43:59.000000000 -0800
+++ Source/cmake/OptionsGTK.cmake	2016-03-08 10:44:29.000000000 -0800
@@ -427,7 +427,6 @@
 # Override the cached variables, gtk-doc and gobject-introspection do not really work when cross-building or when building on Mac.
 if (CMAKE_CROSSCOMPILING OR APPLE)
     set(ENABLE_GTKDOC OFF)
-    set(ENABLE_INTROSPECTION OFF)
 endif ()
 
 set(DERIVED_SOURCES_GOBJECT_DOM_BINDINGS_DIR ${DERIVED_SOURCES_DIR}/webkitdom)
