--- src/tracker-extract/tracker-extract-gif.c.orig	2014-03-22 11:26:52.000000000 +0100
+++ src/tracker-extract/tracker-extract-gif.c	2014-03-22 11:56:42.000000000 +0100
@@ -73,18 +73,16 @@
 	return (GIF_OK);
 }
 
-#if GIFLIB_MAJOR >= 5
+#if (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4
 static inline void
-gif_error (const gchar *action, int err)
+gif_error (const gchar *action)
 {
-	const char *str = GifErrorString (err);
+	const char *str = GifErrorString ();
 	if (str != NULL) {
 		g_message ("%s, error: '%s'", action, str);
-	} else {
-		g_message ("%s, undefined error %d", action, err);
 	}
 }
-#endif /* GIFLIB_MAJOR >= 5 */
+#endif /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
 
 static void
 read_metadata (TrackerSparqlBuilder *preupdate,
@@ -111,22 +109,22 @@
 		ExtBlock extBlock;
 
 		if (DGifGetRecordType(gifFile, &RecordType) == GIF_ERROR) {
-#if GIFLIB_MAJOR < 5
+#if ((GIFLIB_MAJOR == 4 && GIFLIB_MINOR < 2) || GIFLIB_MAJOR < 4)
 			PrintGifError ();
-#else  /* GIFLIB_MAJOR < 5 */
-			gif_error ("Could not read next GIF record type", gifFile->Error);
-#endif /* GIFLIB_MAJOR < 5 */
+#else  /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
+			gif_error ("Could not read next GIF record type");
+#endif /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
 			return;
 		}
 
 		switch (RecordType) {
 			case IMAGE_DESC_RECORD_TYPE:
 			if (DGifGetImageDesc(gifFile) == GIF_ERROR) {
-#if GIFLIB_MAJOR < 5
-				PrintGifError();
-#else  /* GIFLIB_MAJOR < 5 */
-				gif_error ("Could not get GIF record information", gifFile->Error);
-#endif /* GIFLIB_MAJOR < 5 */
+#if ((GIFLIB_MAJOR == 4 && GIFLIB_MINOR < 2) || GIFLIB_MAJOR < 4)
+                PrintGifError ();
+#else  /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
+				gif_error ("Could not get GIF record information");
+#endif /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
 				return;
 			}
 
@@ -136,11 +134,11 @@
 			framedata = g_malloc (framewidth*frameheight);
 
 			if (DGifGetLine(gifFile, framedata, framewidth*frameheight)==GIF_ERROR) {
-#if GIFLIB_MAJOR < 5
-				PrintGifError();
-#else  /* GIFLIB_MAJOR < 5 */
-				gif_error ("Could not load a block of GIF pixes", gifFile->Error);
-#endif /* GIFLIB_MAJOR < 5 */
+#if ((GIFLIB_MAJOR == 4 && GIFLIB_MINOR < 2) || GIFLIB_MAJOR < 4)
+                PrintGifError ();
+#else  /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
+				gif_error ("Could not load a block of GIF pixes");
+#endif /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
 				return;
 			}
 
@@ -616,7 +614,7 @@
 	gchar *filename, *uri;
 	GFile *file;
 	int fd;
-#if GIFLIB_MAJOR >= 5
+#if (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4
 	int err;
 #endif
 
@@ -643,13 +641,13 @@
 		return FALSE;
 	}	
 
-#if GIFLIB_MAJOR < 5
+#if ((GIFLIB_MAJOR == 4 && GIFLIB_MINOR < 2) || GIFLIB_MAJOR < 4)
 	if ((gifFile = DGifOpenFileHandle (fd)) == NULL) {
 		PrintGifError ();
-#else   /* GIFLIB_MAJOR < 5 */
-	if ((gifFile = DGifOpenFileHandle (fd, &err)) == NULL) {
-		gif_error ("Could not open GIF file with handle", err);
-#endif /* GIFLIB_MAJOR < 5 */
+#else  /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
+	if ((gifFile = DGifOpenFileHandle (fd)) == NULL) {
+		gif_error ("Could not open GIF file with handle");
+#endif /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
 		g_free (filename);
 		close (fd);
 		return FALSE;
@@ -671,11 +669,11 @@
 	g_free (uri);
 
 	if (DGifCloseFile (gifFile) != GIF_OK) {
-#if GIFLIB_MAJOR < 5
+#if ((GIFLIB_MAJOR == 4 && GIFLIB_MINOR < 2) || GIFLIB_MAJOR < 4)
 		PrintGifError ();
-#else  /* GIFLIB_MAJOR < 5 */
-		gif_error ("Could not close GIF file", gifFile->Error);
-#endif /* GIFLIB_MAJOR < 5 */
+#else  /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
+		gif_error ("Could not close GIF file");
+#endif /* (GIFLIB_MAJOR == 4 && GIFLIB_MINOR >= 2) || GIFLIB_MAJOR > 4 */
 	}
 
 	return TRUE;
