--- configure.orig	2013-02-27 18:09:34.000000000 -0800
+++ configure	2013-08-30 21:40:56.000000000 -0700
@@ -5856,9 +5856,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
@@ -5931,9 +5931,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
@@ -6006,9 +6006,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
@@ -6081,9 +6081,9 @@
 
 
 
-  PYTHON_PREFIX='${prefix}'
+  PYTHON_PREFIX=`$PYTHON -c 'import sys; print(sys.prefix);'`
 
-  PYTHON_EXEC_PREFIX='${exec_prefix}'
+  PYTHON_EXEC_PREFIX=`$PYTHON -c 'import sys; print(sys.exec_prefix);'`
 
 
 
