--- Makefile.in.orig	2013-02-27 18:09:36.000000000 -0800
+++ Makefile.in	2013-08-30 21:40:56.000000000 -0700
@@ -328,8 +328,8 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = data po Alacarte
-CLEANFILES = alacarte
-bin_SCRIPTS = alacarte
+CLEANFILES = alacarte.py
+bin_SCRIPTS = alacarte.py
 DISTCLEANFILES = ChangeLog
 EXTRA_DIST = \
 	alacarte.in \
@@ -840,11 +840,11 @@
 	tags-am uninstall uninstall-am uninstall-binSCRIPTS
 
 
-alacarte: alacarte.in
+alacarte.py: alacarte.in
 	$(AM_V_GEN)sed -e s!\@PYTHON\@!@PYTHON@!	\
 	    -e s!\@PYOPTIONS\@!-OOt!			\
 	    < $< > $@
-alacarte: Makefile
+alacarte.py: Makefile
 
 ChangeLog:
 	@echo Creating $@
