--- configure.ac.orig	2014-10-13 14:55:17.000000000 -0700
+++ configure.ac	2014-10-13 14:55:40.000000000 -0700
@@ -76,7 +76,7 @@
 dnl ===========================================================================
 
 CFLAGS=${CFLAGS-:""}
-CFLAGS="$CFLAGS -fPIC -DPIC -Wl,--as-needed"
+CFLAGS="$CFLAGS -fPIC -DPIC"
 
 AC_ARG_ENABLE(deprecated,
 	      AS_HELP_STRING([--disable-deprecated], [do not use deprecated API]),
@@ -332,7 +332,7 @@
 
 dnl ===========================================================================
 
-WARN_CFLAGS="-Wall -Wcast-align -Wtype-limits -Wclobbered -Wempty-body -Wignored-qualifiers"
+WARN_CFLAGS="-Wall -Wcast-align -Wtype-limits -Wempty-body -Wignored-qualifiers"
 
 for option in $WARN_CFLAGS; do
 	SAVE_CFLAGS="$CFLAGS"
