--- gio/gdbusprivate.c.orig	2015-05-22 14:01:27.000000000 -0700
+++ gio/gdbusprivate.c	2015-07-20 17:28:35.000000000 -0700
@@ -2068,7 +2068,7 @@
   /* TODO: use PACKAGE_LOCALSTATEDIR ? */
   ret = NULL;
   first_error = NULL;
-  if (!g_file_get_contents ("/var/lib/dbus/machine-id",
+  if (!g_file_get_contents ("/opt/local/var/lib/dbus/machine-id",
                             &ret,
                             NULL,
                             &first_error) &&
@@ -2078,7 +2078,7 @@
                             NULL))
     {
       g_propagate_prefixed_error (error, first_error,
-                                  _("Unable to load /var/lib/dbus/machine-id or /etc/machine-id: "));
+                                  _("Unable to load /opt/local/var/lib/dbus/machine-id or /etc/machine-id: "));
     }
   else
     {
