--- configure.ac.orig	2014-02-14 16:34:10.000000000 +0100
+++ configure.ac	2014-02-14 16:35:39.000000000 +0100
@@ -230,11 +230,9 @@
               AS_HELP_STRING([--enable-gsettings=yes|no],
                              [use gsettings instead of gconf (default is auto)]),
               ENABLE_GSETTINGS=$enableval,
-              AC_CHECK_PROG([HAS_DCONF], [dconf], [yes], [no])
               [PKG_CHECK_EXISTS([giomm-2.4 >= $LIBGIOMM_WITH_GSETTINGS_VERSION gsettings-desktop-schemas >= $GSETTINGS_DESKTOP_SCHEMAS], [HAS_GSETTINGS=yes], [HAS_GSETTINGS=no])])
 if test x$ENABLE_GSETTINGS = xyes || (test x$ENABLE_GSETTINGS = xauto \
-      && test x$HAS_GSETTINGS = xyes \
-      && test x$HAS_DCONF = xyes); then
+      && test x$HAS_GSETTINGS = xyes); then
     AC_DEFINE([WITH_GSETTINGS], 1, [build with gsettings instead of gconf])
     AC_DEFINE([CONFIG_MGR_MODULE_NAME], ["gsettingsmgr"], [configuration manager to use])
     CONF_MGR=gsettings
