--- Source/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp.orig	2013-11-28 09:11:39.000000000 -0800
+++ Source/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp	2013-11-28 09:12:14.000000000 -0800
@@ -39,8 +39,8 @@
 #include "TextEncoding.h"
 #include <cairo-ft.h>
 #include <cairo.h>
-#include <freetype/freetype.h>
-#include <freetype/tttables.h>
+#include <freetype2/freetype.h>
+#include <freetype2/tttables.h>
 #include <hb.h>
 #include <wtf/text/CString.h>
 
