--- gdata/gdata.symbols.orig	2015-07-07 05:53:12.000000000 -0700
+++ gdata/gdata.symbols	2015-07-11 09:35:57.000000000 -0700
@@ -1095,8 +1095,6 @@
 gdata_freebase_topic_result_get_type
 gdata_freebase_topic_result_new
 gdata_freebase_topic_result_dup_object
-gdata_freebase_result_error_get_type
-gdata_freebase_result_error_quark
 gdata_freebase_result_get_type
 gdata_freebase_result_new
 gdata_freebase_result_dup_variant
